astropy/cosmology/tests/test_core.py

```diff
@@ -31,16 +31,22 @@ class Example1(Cosmology):
                               unit=u.m, equivalencies=u.mass_energy())
 
             def __init__(self, param=15):
-                self._param = param
+                self.param = param
 
+        #  with getter and validator
         class Example2(Example1):
             def __init__(self, param=15 * u.m):
-                self._param = param.to_value(u.km)
+                self._param = self.__class__.param.validate(self, param)
```

Comment:
Becoming more and more convinced one should not limit `__set__`...

Reply:
Agreed on allowing both ``getter`` and ``__set__``.